### ISSUE TITLE
Error about  to_grib

### ISSUE THREAD

**Issue by lcrystallc** [ecmwf/cfgrib#267]
when I run the code`ds = xr.open_dataset('D:/work/data/era5-levels-members.grib', engine='cfgrib')
to_grib(ds, 'D:/work/data/model_data/out1.grib', grib_keys={'edition': 2})`, it retuen the error: File "C:\ProgramData\Anaconda3\envs\spyder_2\lib\site-packages\pandas\core\indexes\datetimes.py", line 700, in get_loc
    raise KeyError(key)

KeyError: 1483228800000000000
But I can import pandas well , and I can't find  is there any problem with pands  

I don't know what I need to do to get past this issue. Thanks for anyone help me ~

**Comment by ChristopheLRTE** [comment]
Hello @lcrystallc,

I'm not an expert, but in my opinion it's related to this issue https://github.com/pydata/xarray/pull/4292 (_I found this link into the code of xarray/core/indexes.py_).

And I tried to update the code (_precisely in indexes.py, class **PandasIndex**, method **query()**_) like that (_look at the line just before my comment in the code below_), and it's ok for me :
```python
def query(
        self, labels, method=None, tolerance=None
    ) -> Tuple[Any, Union["PandasIndex", None]]:
        assert len(labels) == 1
        coord_name, label = next(iter(labels.items()))
        index = self.array

        if isinstance(label, slice):
            indexer = _query_slice(index, label, coord_name, method, tolerance)
        elif is_dict_like(label):
            raise ValueError(
                "cannot use a dict-like object for selection on "
                "a dimension that does not have a MultiIndex"
            )
        else:
            label = (
                label
                if getattr(label, "ndim", 1) > 1  # vectorized-indexing
                else _asarray_tuplesafe(label)
            )
            if label.ndim == 0:
                # see https://github.com/pydata/xarray/pull/4292 for details
                label_value = label[()] if label.dtype.kind in "mM" else label.item()
                
                if isinstance(index, pd.CategoricalIndex):
                    if method is not None:
                        raise ValueError(
                            "'method' is not a valid kwarg when indexing using a CategoricalIndex."
                        )
                    if tolerance is not None:
                        raise ValueError(
                            "'tolerance' is not a valid kwarg when indexing using a CategoricalIndex."
                        )
                    indexer = index.get_loc(label_value)
                else:
                    indexer = index.get_loc(
                        self.array[0], method=method, tolerance=tolerance
#                       ^^^^^^^^^^^^^      instead of 'label_value, method=method, tolerance=tolerance'
                    )
            elif label.dtype.kind == "b":
                indexer = label
            else:
                indexer = get_indexer_nd(index, label, method, tolerance)
                if np.any(indexer < 0):
                    raise KeyError(f"not all values found in index {coord_name!r}")

        return indexer, None
```
Hope this helps,
Christophe

**Comment by alexamici** [comment]
Fixed in #272
